pcs_test/tier0/lib/cib/resource/test_stonith.py

```diff
@@ -279,3 +286,140 @@ def test_unsupported_provider(self):
+            '<primitive id="fence-mepath" class="stonith" type="fence_mpath" />'
+        )
+    return etree.fromstring(
+        f"""
+        <primitive id="fence-mepath" class="stonith" type="fence_mpath">
+            <instance_attributes id="fence-mepath-instance_attributes">
+                <nvpair id="fence-mepath-instance_attributes-pcmk_host_map"
+                    name="pcmk_host_map" value="{host_map_value}" />
+            </instance_attributes>
+        </primitive>
+        """
+    )
+
+
+class GetNodeKeyMapForMpath(TestCase):
+    NODE_LABELS = ["rh9-1", "rh9-2", "rh9-3"]
+
+    @staticmethod
+    def fixture_error(missing_nodes=None, host_map_value=""):
+        if not missing_nodes:
+            reason = (
+                "mpath reservation keys not found in 'pcmk_host_map' attribute"
+            )
+        else:
+            keys = format_plural(missing_nodes, "key")
+            nodes = format_plural(missing_nodes, "node")
+            node_list = format_list(missing_nodes)
+            reason = (
+                f"missing mpath reservation {keys} for {nodes}: {node_list}, in"
+                f" value '{host_map_value}'"
+            )
+        return fixture.error(
+            reports.codes.STONITH_RESTARTLESS_UPDATE_UNABLE_TO_PERFORM,
+            reason=reason,
+            reason_type="other",
+        )
+
+    def assert_success(self, value, result_dict, node_labels=None):
+        if node_labels is None:
+            node_labels = self.NODE_LABELS
+        self.assertEqual(
+            stonith.get_node_key_map_for_mpath(
+                _fixture_stonith_el(value), node_labels
+            ),
+            result_dict,
+        )
+
+    def assert_error(self, value, node_labels=None, missing_nodes=None):
+        if node_labels is None:
+            node_labels = self.NODE_LABELS
+        assert_raise_library_error(
+            lambda: stonith.get_node_key_map_for_mpath(
+                _fixture_stonith_el(value), node_labels
+            ),
+            self.fixture_error(
+                missing_nodes=missing_nodes, host_map_value=value
+            ),
+        )
+
+    def test_invalid_part_skipped(self):
+        self.assert_success(
+            "  ;\trh9-1:1;rh9-2==22;rh9-3 :3;e",
+            {"rh9-1": "1"},
+            node_labels=self.NODE_LABELS[0:1],
+        )
+
+    def test_success_colon_sign_as_assigment_char(self):
+        self.assert_success(
+            "rh9-1:1;rh9-2:2;rh9-3:3",
+            {"rh9-1": "1", "rh9-2": "2", "rh9-3": "3"},
+        )
+
+    def test_success_equal_sign_as_assigment_char(self):
+        self.assert_success(
+            "rh9-1=1;rh9-2=2;rh9-3=3",
+            {"rh9-1": "1", "rh9-2": "2", "rh9-3": "3"},
+        )
+
+    def test_success_mixed_assigment_char(self):
+        self.assert_success(
+            "rh9-1:1;rh9-2=2;rh9-3:3",
+            {"rh9-1": "1", "rh9-2": "2", "rh9-3": "3"},
+        )
+
+    def test_scucess_semicolon_as_separator(self):
```

Comment:
```suggestion
    def test_success_semicolon_as_separator(self):
```
The same typo is in tests below as well.